src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.Fixer.cs

```diff
@@ -5,11 +5,136 @@
 using Microsoft.QualityGuidelines.CSharp.Analyzers;
 using Microsoft.QualityGuidelines.VisualBasic.Analyzers;
 using Test.Utilities;
+using Xunit;
 
 namespace Microsoft.QualityGuidelines.Analyzers.UnitTests
 {
     public class UseLiteralsWhereAppropriateFixerTests : CodeFixTestBase
     {
+        [Fact]
+        public void CSharp_CodeFixForEmptyString()
+        {
+            VerifyCSharpFix(@"
+class C
+{
+    public /*leading*/ static /*intermediate*/ readonly /*trailing*/ string f1 = """";
+}
+",
+                @"
+class C
+{
+    public /*leading*/ const /*intermediate*/  /*trailing*/ string f1 = """";
+}
+");
+
+            VerifyBasicFix(@"
+Class C
+    Public Shared ReadOnly f1 As String = """"
+End Class
+",
+@"
+Class C
+    Public Const f1 As String = """"
+End Class
+");
+        }
+
+        [Fact]
+        public void CSharp_CodeFixForNonEmptyString()
+        {
+            VerifyCSharpFix(@"
+class C
+{
+    /*leading*/
+    readonly /*intermediate*/ static /*trailing*/ string f1 = ""Nothing"";
+}
+",
+                @"
+class C
+{
+    /*leading*/
+    const /*intermediate*/  /*trailing*/ string f1 = ""Nothing"";
+}
+");
+
+            VerifyBasicFix(@"
+Class C
+    'leading
+    ReadOnly Shared f1 As String = ""Nothing""
+End Class
+",
+@"
+Class C
+    'leading
+    Const f1 As String = ""Nothing""
+End Class
+");
+        }
+
+        [Fact]
+        public void CSharp_CodeFixForMultiDeclaration()
+        {
+            VerifyCSharpFix(@"
+class C
+{
+    /*leading*/
+    readonly /*intermediate*/ static /*trailing*/ string f3, f4 = ""Message is shown only for f4"";
+}
+",
+                @"
+class C
+{
+    /*leading*/
+    const /*intermediate*/  /*trailing*/ string f3, f4 = ""Message is shown only for f4"";
+}
+");
+
+            VerifyBasicFix(@"
+Class C
+    'Error BC30438: Constants must have a value.
+    Shared ReadOnly f3 As String, f4 As String = ""Message is shown only for f4""
+End Class
+",
+@"
+Class C
+    'Error BC30438: Constants must have a value.
+    Const f3 As String, f4 As String = ""Message is shown only for f4""
+End Class
+", validationMode: TestValidationMode.AllowCompileErrors, allowNewCompilerDiagnostics: true);
```

Comment:
I'm having second thoughts here, about whether introducing a build error is the right solution. An alternative would be to suppress the fixer in this case. Any guidance would be appreciated.